### PR TITLE
Fix dist stats when fill_gaps = TRUE

### PR DESCRIPTION
**Pull request by Aeilert**
Hi @tonyfujs 

I looked into this issue originally raised by @danielmahler.

>  Why are the missing value patterns for gini and median different?  I think it should follow the same missing value pattern as the Gini

I think this was caused by how merging between the output of `fg_pip()` was conducted. Using `country_code`, `reporting_year` etc. doesn't work because these are not linked to observations in the dist stat table. For extrapolated values we would like to match e.g AGO 1981 (ref year) with AGO 2000 (survey year) but using the  `reporting_year` for this merge (obviously)  doesn't work. However using the `cache_id` will since the cache_id used for extrapolation (AGO_2000_xxx) is the same as the corresponding survey / cache_id in the dist stat table. 

To get a better sense of the issue try: 

```r
u <- "https://api.worldbank.org/pip/v1/pip?country=all&year=all&povline=1.9&fill_gaps=TRUE"
df <- jsonlite::fromJSON(u)
df <- df[(is.na(df$gini) | is.na(df$median)),]
df <- df[!(is.na(df$gini) & is.na(df$median)),]
df[c('gini', 'median')]

pip("AGO", year = 1981, fill_gaps = TRUE, lkup = lkups$versions_paths$latest_release)
pip("AGO", year = 2001, fill_gaps = TRUE, lkup = lkups$versions_paths$latest_release)
pip("AGO", year = "all",  fill_gaps = TRUE, lkup = lkups)
``` 

Let me know if you have questions.

### CODE REVIEW

**Review by tonyfujs — COMMENTED** [review]
Hey @Aeilert 
I'm getting 15 unit test failure on this branch. Should I pulled the most recent data from TFS for this, or is there something else going one? Are the tests running for you?
Thanks!

File: R/fg_pip.R [hunk 1]
```diff
@@ -99,21 +99,28 @@ fg_pip <- function(country,
   out <- unlist(out, recursive = FALSE)
   out <- data.table::rbindlist(out)
 
+  # Set cache_id to a modified version of data_interpolation_id
+  out$cache_id <-
+    ifelse(grepl("[|]", out$data_interpolation_id),
+           gsub(paste0("_", out$reporting_level, collapse = '|'), '', out$data_interpolation_id),
+           out$cache_id)
+
```
[the remark this answers is not in the document]
Comment (Aeilert) [reply R/fg_pip.R]: Done.

**Comment by Aeilert** [comment]
> Hey @Aeilert I'm getting 15 unit test failure on this branch. Should I pulled the most recent data from TFS for this, or is there something else going one? Are the tests running for you? Thanks!

Hey @tonyfujs. You are right. Some of the new unit tests had a typo. And there was a also a new bug for the merge w/ dists stats when fill_gaps = FALSE and reporting_level = national. This has now been fixed. Let me know if you still find errors.

**Comment by tonyfujs** [comment]
Thanks @Aeilert 
I still have two unit tests failing:
- One is due to a known issue that is being tackled separately, so no problem
- The other one I'm not sure. It happends in `test-pip-local`. It is not even a unit test failing, but an code error...
```
test_that("Imputation is working", {
  tmp <- pip(
    country = "all",
    year = "all",
    povline = 3.5,
    fill_gaps = TRUE,
    lkup = lkups
  )
  Expect there are no duplicates
  expect_equal(nrow(unique(tmp[, c("country_code",
                                   "reporting_year",
                                   "reporting_level",
                                   "welfare_type")])),
               nrow(tmp))
 
})
```

**Comment by Aeilert** [comment]
> Thanks @Aeilert I still have two unit tests failing:
> 
> * One is due to a known issue that is being tackled separately, so no problem
> * The other one I'm not sure. It happends in `test-pip-local`. It is not even a unit test failing, but an code error...
> 
> ```
> test_that("Imputation is working", {
>   tmp <- pip(
>     country = "all",
>     year = "all",
>     povline = 3.5,
>     fill_gaps = TRUE,
>     lkup = lkups
>   )
>   Expect there are no duplicates
>   expect_equal(nrow(unique(tmp[, c("country_code",
>                                    "reporting_year",
>                                    "reporting_level",
>                                    "welfare_type")])),
>                nrow(tmp))
>  
> })
> ```

Thanks Tony. Bad coding on my part.  Fixed by using `paste0("_", unique(out$reporting_level), collapse = '|')`  instead of paste0("_", out$reporting_level, collapse = '|'). Also update the NEWS.md.

**Review by tonyfujs — APPROVED** [review]
Thanks @Aeilert 
Everything looks good now!

**Pull request merged**